### ISSUE TITLE
instruct模型是否支持模型微调

### ISSUE THREAD

**Issue by Strive-for-excellence** [FunAudioLLM/CosyVoice#230]
如题

**Comment by aluminumbox** [comment]
yes,, but you need to provide instruct data, which is relatively difficult to acquire

**Comment by Strive-for-excellence** [comment]
> yes,, but you need to provide instruct data, which is relatively difficult to acquire

用run.sh的脚本就能训练instruct吗，还是需要自己修改train.py的代码

**Comment by aluminumbox** [comment]
> > yes,, but you need to provide instruct data, which is relatively difficult to acquire
> 
> 非常感谢回复！用run.sh的脚本就能训练instruct吗，还是需要自己修改train.py的代码？

no need to modify train code, you only need to change the text in parquet file to something like instruct_text + <endofprompt> + tts_text, check frontend.py for instruct details

**Comment by yangyazhou97** [comment]
I constructed the training data as you suggested to finetune the CosyVoice-300M model, but the prompt between startofprompt and endofprompt still can be read out.
cosyvoice.yaml :
`use_spk_embedding: True,
lr: 1e-5,
scheduler: constantlr`
Here is my training data:
![image](https://github.com/user-attachments/assets/145c6f40-c0b2-41ac-a39e-5841e4c37b06)

**Comment by Chi8wah** [comment]
请教一下，楼主目前有 instruct 的微调进展吗

**Comment by fengping1** [comment]
instruct 微调后，模型推理不正常，提前预测出4096， 这个会是什么问题

**Comment by aluminumbox** [comment]
> instruct 微调后，模型推理不正常，提前预测出4096， 这个会是什么问题

看一下Instruct推理方式，llm是没有embedding的，按照instruct的推理方式构造训练的数据
